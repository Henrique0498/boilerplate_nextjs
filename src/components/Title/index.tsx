interface InTitle extends React.ComponentPropsWithRef<'h1'> {}

const Title = ({ children, ...props }: InTitle) => {
  return <h1 {...props}>{children}</h1>
}
export default Title
