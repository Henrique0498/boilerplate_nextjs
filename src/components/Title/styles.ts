'use client'

import styled from 'styled-components'

export const TitleStyle = styled.h1`
  font-size: 3rem;
`
