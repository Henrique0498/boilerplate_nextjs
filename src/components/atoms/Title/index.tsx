import * as S from './styles'

interface InTitle extends React.ComponentPropsWithRef<'h1'> {}

const Title = ({ children, ...props }: InTitle) => {
  return <S.TitleStyle {...props}>{children}</S.TitleStyle>
}
export default Title
