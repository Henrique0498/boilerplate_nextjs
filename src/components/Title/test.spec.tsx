import { render, screen } from '@testing-library/react'

import Title from '.'

describe('<Title />', () => {
  it('should render the children', () => {
    render(<Title>Testando...</Title>)

    expect(
      screen.getByRole('heading', { name: /testando/i })
    ).toBeInTheDocument()
  })
})
