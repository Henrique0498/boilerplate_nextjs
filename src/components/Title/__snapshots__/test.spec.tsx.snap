// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> should render the children 1`] = `
.c0 {
  font-size: 3rem;
}

<h1
  class="c0"
>
  Testando...
</h1>
`;
